DataFormats/Provenance/interface/BranchPattern.h

```diff
@@ -0,0 +1,66 @@
+#ifndef DataFormats_Provenance_interface_BranchPattern_h
+#define DataFormats_Provenance_interface_BranchPattern_h
+
+#include <string>
+#include <regex>
+#include <vector>
+
+#include "DataFormats/Provenance/interface/ProductDescription.h"
+
+namespace edm {
+
+  /* BranchPattern
```

Comment:
So we are making a concerted effort to replace the use of 'Branch' in the framework (as it really only refers to TBranch from ROOT and with RNTuple that will go away). This is really more of a `ProductLookupPattern` or `ProductNamingPattern` (as items in an Event are formally 'Data Products' in the framework).

Reply:
No problem, I can rename it to `ProductLookupPattern` or `ProductNamingPattern`.